etc/node_ws/app.js

```diff
@@ -0,0 +1,98 @@
+// Defaults; configurable through environment
+const HOST = process.env.HOST || '127.0.0.1';
+const PORT = process.env.PORT || 8080;
+const SUB_PATH = process.env.SUB_PATH || '/subscribe/'
+const CLUSTER_ID = process.env.CLUSTER_ID || 'usgs'
+const CHANNEL = process.env.CHANNEL || 'anss.pdl.realtime'
+const PING_INTERVAL = process.env.PING_INTERVAL || 30000;
+const LOG_LEVEL = process.env.LOG_LEVEL || 'info';
+
+var package = require('./package.json')
+console.log('Starting ' + package.name + ' version ' + package.version);
+
+// stan setup
+const stan = require('node-nats-streaming');
+
+// web socket setup
+const WebSocket = require('ws');
+const server = new WebSocket.Server({host:HOST, port:PORT});
+
+// uuid
+const { uuid } = require('uuidv4');
+
+console.log('Setup complete, waiting for connections...');
+
+server.on('connection', function onConnection(ws, req) {
```

Comment:
Refactor this into 3 parts:

- An "App" class/function that accepts options, starts the server, accepts connections, and handles pings/pongs.
- A "Handler" class/function that handles one connection
- A "main" function that reads configuration from the environment and starts the App